### PR TITLE
Fix map distortion due to high latitudes

### PR DESCRIPTION
**Pull request by jkflying**
**Describe problem solved by this pull request**
Current formula used for distance in the rally point calculations didn't take map distortion into account.

**Describe your solution**
Use a local correction for the longitude compression

**Describe possible alternatives**
Use the 'real' geo library distance. However this is slow, and overkill IMO

### CODE REVIEW

**Comment by dagar** [comment]
I think I'd still personally prefer leaning on the geo library for low frequency work like this, but whichever you prefer.

I'll merge as soon as flash is available.

**Comment by jkflying** [comment]
I think the worry is that we might have a whole bunch of rally points (maybe from a database?), so despite it running infrequently it still causes a big pause. I guess this was the concern of @sfuhrer in just using the lat/lon directly. Otherwise we could add this method to the geo library as a `_squared_fast` variant (maybe similar to fast trig approximations), but then it won't share the `cos(lat)` call across the points.

File: src/modules/navigator/rtl.cpp [hunk 1]
```diff
@@ -90,7 +90,15 @@ void RTL::find_RTL_destination()
 	// get distance to home position
 	double dlat = home_landing_position.lat - global_position.lat;
 	double dlon = home_landing_position.lon - global_position.lon;
-	double min_dist_squared = dlat * dlat + dlon * dlon;
+
+	double lon_scale = cos(global_position.lat * (M_PI / 180));
```
Comment (MaEtUgR) [inline src/modules/navigator/rtl.cpp:94]:
```suggestion
	double lon_scale = cos(global_position.lat * (M_PI / 180.));
```

File: src/modules/navigator/rtl.cpp [hunk 2]
```diff
@@ -90,7 +90,15 @@ void RTL::find_RTL_destination()
...
+
+	auto coord_dist_sq = [lon_scale](double lat_diff, double lon_diff) -> double {
+		double lon_diff_scaled =  lon_scale * matrix::wrap<double>(lon_diff, -180, 180);
```
Comment (MaEtUgR) [inline src/modules/navigator/rtl.cpp:97]:
```suggestion
		double lon_diff_scaled =  lon_scale * matrix::wrap<double>(lon_diff, -180., 180.);
```

File: src/modules/navigator/rtl.cpp (hunk 1 above)
Comment (MaEtUgR) [inline src/modules/navigator/rtl.cpp:94]:
```suggestion
	double lon_scale = cos(math::radians(global_position.lat));
```

**Review by MaEtUgR — APPROVED** [review]
Thanks for the improvements 👍

**Pull request merged**